### PR TITLE
Update dependencies except espresso (2017-02-25)

### PR DESCRIPTION
**Pull request by ryanmoelter**
The dependencies are getting a little stale and I got tired of android studio complaining, so I updated them all to the latest versions as of today.

I left out espresso because it had some breaking changes in 3.0 and I don't know enough about it to know the potential repercussions.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,10 +4,13 @@ android:
...
     - tools
     - platform-tools
-    - build-tools-25.0.1
-    - android-25
+    - build-tools-27.0.2
+    - android-27
     - extra
 
+before_install:
+  - yes | sdkmanager "platforms;android-27"
```
Comment (ryanmoelter) [inline .travis.yml]: This is to work around an issue with Travis CI: https://github.com/travis-ci/travis-ci/issues/8874

File: magellan-support/src/main/java/com/wealthfront/magellan/support/SingleActivity.java [hunk 2]
```diff
@@ -34,6 +34,7 @@ protected void onPostCreate(@Nullable Bundle savedInstanceState) {
   @Override
   public void onSaveInstanceState(Bundle outState) {
     navigator.onSaveInstanceState(outState);
+    super.onSaveInstanceState(outState);
```
Comment (ryanmoelter) [inline magellan-support/src/main/java/com/wealthfront/magellan/support/SingleActivity.java]: Sneaky fix for #42 (also fixes a new lint error)

File: .idea/misc.xml [hunk 3]
```diff
@@ -27,17 +27,7 @@
...
-    <OptionsSetting value="true" id="Remove" />
-    <OptionsSetting value="true" id="Checkout" />
-    <OptionsSetting value="true" id="Update" />
-    <OptionsSetting value="true" id="Status" />
-    <OptionsSetting value="true" id="Edit" />
-    <ConfirmationsSetting value="0" id="Add" />
-    <ConfirmationsSetting value="0" id="Remove" />
-  </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_7" default="true" assert-keyword="true" jdk-15="true" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_7" default="true" project-jdk-name="1.8 (2)" project-jdk-type="JavaSDK">
```
Comment (fabiendevos) [inline .idea/misc.xml]: revert?

File: .travis.yml (hunk 1 above)
Comment (fabiendevos) [reply .travis.yml]: maybe other workaround...

File: magellan-support/src/main/java/com/wealthfront/magellan/support/SingleActivity.java (hunk 2 above)
Comment (fabiendevos) [reply magellan-support/src/main/java/com/wealthfront/magellan/support/SingleActivity.java]: I'll take it

File: .idea/misc.xml (hunk 3 above)
Comment (ryanmoelter) [reply .idea/misc.xml]: Done

**Pull request merged**